migrate.lua

```diff
@@ -80,20 +80,114 @@ local function migrate_v2_to_v3()
 	end)
 end
 
+
+
+local function search_box(playername, box, uuid)
+	local e = mail.get_storage_entry(playername)
+	for _, m in ipairs(e[box]) do
+		if m.id == uuid then
+		return { time = m.time, from = m.from, to = m.to, cc = m.cc, bcc = m.bcc, subject = m.subject, body = m.body } end
+	end
+	return false
+end
+
+local function is_uuid_existing(uuid)
+    for _, k in ipairs(mail.storage:get_keys()) do
+        if string.sub(k,1,5) == "mail/" then
+			local p = string.sub(k, 6)
+			local result
+			local boxes = {"inbox", "outbox", "drafts", "trash"}
+			for _, b in ipairs(boxes) do
+				result = search_box(p, b, uuid)
+				if result then return result end
+			end
+		end
+    end
+    return false
+end
+
+local function are_message_sames(a, b)
+	return a.time == b.time
+	   and a.from == b.from
+	   and a.to == b.to
+	   and a.cc == b.cc
+	   and a.bcc == b.bcc
+	   and a.subject == b.subject
+	   and a.body == b.body
+end
+
+local function repair_box(playername, box)
+	local e = mail.get_storage_entry(playername)
+	for _, m in ipairs(e[box]) do
+		local uuid = m.id
+		local exists = is_uuid_existing(uuid)
+		if exists and not are_message_sames(exists, m) then
+			local new_uuid = mail.new_uuid() -- generates a new uuid to replace doublons
+			for _, k in ipairs(mail.storage:get_keys()) do
+				if string.sub(k,1,5) == "mail/" then
+					local p = string.sub(k, 6)
+					local er = mail.get_storage_entry(p)
+					for _, r in ipairs(er.inbox) do
+						if r.id == uuid and not are_message_sames(m, r) then
+							r.id = new_uuid
+						end
+					end
+					for _, r in ipairs(er.outbox) do
+						if r.id == uuid and not are_message_sames(m, r) then
+							r.id = new_uuid
+						end
+					end
+					for _, r in ipairs(er.drafts) do
+						if r.id == uuid and not are_message_sames(m, r) then
+							r.id = new_uuid
+						end
+					end
+					for _, r in ipairs(er.trash) do
+						if r.id == uuid and not are_message_sames(m, r) then
+							r.id = new_uuid
+						end
+					end
+					mail.set_storage_entry(p, er)
+				end
+			end
+		end
+	end
+end
+
+-- repair database for uuid doublons
+local function repair_storage()
+	-- iterate through players
+	for _, k in ipairs(mail.storage:get_keys()) do
+		if string.sub(k,1,5) == "mail/" then
+			local p = string.sub(k, 6)
+			repair_box(p, "inbox")
+			repair_box(p, "outbox")
+			repair_box(p, "drafts")
+			repair_box(p, "trash")
```

Comment:
Related to previous, minor naming thing
```suggestion
			fix_duplicate_uuids(p, "inbox")
			fix_duplicate_uuids(p, "outbox")
			fix_duplicate_uuids(p, "drafts")
			fix_duplicate_uuids(p, "trash")
```

Reply:
Done :heavy_check_mark: